packages/squiggle-lang/src/utility/SDate.ts

```diff
@@ -0,0 +1,106 @@
+import { REOther } from "../errors/messages.js";
+import { SDuration, durationUnits } from "./SDuration.js";
+import { Ok, result, Err } from "./result.js";
+import * as Result from "./result.js";
+
+//This is our own internal date class, which is a wrapper around the built-in Date class. It's used by the interpreter, but meant to act like a simple date library.
+export class SDate {
+  constructor(public value: Date) {
+    this.value = value;
+  }
+
+  private static dateIsValid(date: Date): boolean {
+    return date instanceof Date && isFinite(date.getTime());
+  }
+
+  static fromString(str: string): result<SDate, string> {
```

Comment:
Note: Some of these produce results, some errors, which is a bit awkward.

Reply:
How about this uses a ``result`` and returns an error than? We do that in the Symbolic dist lib, results seem good to me.